### ISSUE TITLE
[module_name] "Add PCL grabber for ZED Stereo Camera"

### ISSUE THREAD

**Issue by shrijitsingh99** [PointCloudLibrary/pcl#3758]
ZED Stereo cameras have become quite popular and used by quite a lot of people. Adding its support would be a great feature to have.

**Context**

Currently, OpenNI and RealSense devices are only supported by PCL. It makes sense to have ZED too, it's as popular as the above two. It would be great if we could have a standard API to access all these devices. Usage of these devices through this API would reduce redundancy and help avoid bad practices that are inherent when users try to write their own grabber.

**Expected behavior**

Similar to current implementations of OpenNI and RealSense devices. ZED already has [examples](https://github.com/stereolabs/zed-examples/blob/master/depth%20sensing/cpp/src/main.cpp) to get started with this.

**Current Behavior**

Doesn't exist. Have to write your own grabber using ZED API.

**Describe the solution you'd like**

A wrapper over ZED API having an interface similar to wrappers for existing devices.

**Describe alternatives you've considered**

One question that comes to mind is slightly opposite to the proposed idea is that, do such wrappers need to exist or should be removed from PCL and be left to the respective manufacturers to provide interfaces and users write their own wrappers, instead of bloating PCL. 
Maintaining such a codebase is also an issue as it requires access to physical hardware.

**Comment by larshg** [comment]
Seems like Stereo laps has already done [this](https://github.com/stereolabs/zed-pcl)

**Comment by larshg** [comment]
They also have an [example using PCL](https://github.com/stereolabs/zed-pcl).

**Comment by shrijitsingh99** [comment]
> They also have an [example using PCL](https://github.com/stereolabs/zed-pcl).

Yeah, shouldn't require much effort to build a similar interface like for other devices.

**Comment by SergioRAgostinho** [comment]
In general I like to get some notion of how popular some of these cameras are among our user base before adding grabber implementations. How about we start a "head count" by having interested users "like" the OP?

It's the first time I hear about ZED stereo cameras. Are there any public robotics/computer vision datasets acquired with these?

**Comment by aditya369007** [comment]
I use a ZED camera. The examples that they have provided are good and works well. Is there any advantage of adding direct support in PCL itself?
